src/components/diff/NBDiffHeader.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { IDiffContext, IGitRef, ISpecialRef } from './model';
+
+export interface INBDiffHeaderProps {
+  path: string;
+  diffContext: IDiffContext;
+}
+
+/**
+ * A React component to render the header which shows metadata around the diff
+ * being rendered. Shows the path to the file and the previous and current ref
+ * being used for the diff.
+ */
+export class NBDiffHeader extends React.Component<INBDiffHeaderProps, {}> {
```

Comment:
This should also be a functional component

Reply:
Done!